core/tests/switch_account_tier_test.rs

```diff
@@ -0,0 +1,229 @@
+            ),
+            (
+                test_credit_cards::decline::EXPIRED_CARD,
+                SwitchAccountTierError::CardDeclined(CardDeclineReason::ExpiredCard),
+            ),
+            (
+                test_credit_cards::decline::INCORRECT_NUMBER,
+                SwitchAccountTierError::CardDeclined(CardDeclineReason::IncorrectNumber),
+            ),
+            (
+                test_credit_cards::decline::INCORRECT_CVC,
+                SwitchAccountTierError::CardDeclined(CardDeclineReason::IncorrectCVC),
+            ),
+        ];
+
+        for (card_number, error) in scenarios {
+            let result = api_service::request(
+                &account,
+                SwitchAccountTierRequest {
+                    account_tier: generate_monthly_account_tier(card_number, None, None, None),
+                },
+            );
+
+            assert_matches!(
+                result,
+                Err(ApiError::<SwitchAccountTierError>::Endpoint(error))
+            );
+        }
+    }
+
+    #[test]
+    fn switch_account_tier_invalid_card() {
+        let account = generate_account();
+        let (root, _) = generate_root_metadata(&account);
+        api_service::request(&account, NewAccountRequest::new(&account, &root)).unwrap();
+
+        let scenarios = vec![
+            (
+                test_credit_cards::INVALID_NUMBER,
+                None,
+                None,
+                None,
+                SwitchAccountTierError::InvalidCreditCard(CreditCardRejectReason::Number),
+            ),
+            (
+                test_credit_cards::GOOD,
+                Some("1970"),
+                None,
+                None,
+                SwitchAccountTierError::InvalidCreditCard(CreditCardRejectReason::ExpYear),
+            ),
+            (
+                test_credit_cards::GOOD,
+                None,
+                Some("14"),
+                None,
+                SwitchAccountTierError::InvalidCreditCard(CreditCardRejectReason::ExpMonth),
+            ),
+            (
+                test_credit_cards::GOOD,
+                None,
+                None,
+                Some("11"),
+                SwitchAccountTierError::InvalidCreditCard(CreditCardRejectReason::CVC),
+            ),
+        ];
+
+        for (card_number, maybe_exp_year, maybe_exp_month, maybe_cvc, error) in scenarios {
+            let result = api_service::request(
+                &account,
+                SwitchAccountTierRequest {
+                    account_tier: generate_monthly_account_tier(
+                        card_number,
+                        maybe_exp_year,
+                        maybe_exp_month,
+                        maybe_cvc,
+                    ),
+                },
+            );
+
+            assert_matches!(
+                result,
+                Err(ApiError::<SwitchAccountTierError>::Endpoint(error))
+            );
+        }
+    }
+}
```

Comment:
The case missing here is:

+ create an account
+ use up a bunch of space
+ switch back to free